src/Elastic.Markdown/Myst/InlineParsers/DiagnosticLinkInlineParser.cs

```diff
@@ -275,12 +275,12 @@ private static void ValidateAnchor(InlineProcessor processor, MarkdownFile markd
 			processor.EmitError(link, $"`{anchor}` does not exist in {markdown.FileName}.");
 	}
 
-	private static void UpdateLinkUrl(LinkInline link, string url, ParserContext context, string? anchor, IFileInfo file)
+	private static void UpdateLinkUrl(LinkInline link, string url, ParserContext context, string? anchor)
 	{
 		var urlPathPrefix = context.Build.UrlPathPrefix ?? string.Empty;
 
 		if (!url.StartsWith('/') && !string.IsNullOrEmpty(url))
-			url = GetRootRelativePath(context, file);
+			url = context.CurrentUrlPath[urlPathPrefix.Length..].TrimEnd('/') + '/' + url;
```

Comment:
Lets rely on: 

` Path.GetFullPath(Path.Combine(prefix, path));`

To resolve to rooted url

It solves a lot of edge-cases: https://dotnetfiddle.net/AiCdBc